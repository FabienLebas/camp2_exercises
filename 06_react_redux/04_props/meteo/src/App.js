import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import getWeatherForecastFromCoordinates from './forecast.js';
import getCurrentWeatherFromCoordinates from './current.js'
import getCityName from './google.js';
import HoursInput from './HoursInput.js';

const latitude = 50.6410414;
const longitude = 3.1380786;

class App extends Component {
  constructor(props){
    super(props);
    this.state = {
      morning: "8",
      afternoon: "18",
      city: "Loading your location",
      current: "Loading current weather",
      forecast: "Loading weather forecast"
    }
  }

  handleInputMorning = (input) => {
    this.setState({
      morning: input,
      afternoon: this.state.afternoon,
      city: this.state.city,
      current: this.state.current,
      forecast: this.state.forecast
    });
  }

  handleInputAfternoon = (input) => {
    this.setState({
      morning: this.state.morning,
      afternoon: input,
      city: this.state.city,
      current: this.state.current,
      forecast: this.state.forecast
    });
  }

  componentDidMount(){
    getCityName(latitude, longitude)
      .then(returnedCity => {
        this.setState({
          morning: this.state.morning,
          afternoon: this.state.afternoon,
          city: returnedCity.city,
          current: this.state.current,
          forecast: this.state.forecast
        })
      });
    getCurrentWeatherFromCoordinates(latitude, longitude)
      .then(currentWeather => {
        this.setState({
          morning: this.state.morning,
          afternoon: this.state.afternoon,
          city: this.state.city,
          current: currentWeather,
          forecast: this.state.forecast
        })
      })
    getWeatherForecastFromCoordinates(latitude, longitude, this.state.morning, this.state.afternoon)
      .then(forecastResult => {
        console.log("Initial");
        console.log(forecastResult.map(element => {
          return {
            time: element.FCTTIME.pretty,
            temp: element.temp.metric
          }
        } ));
        this.setState({
          morning: this.state.morning,
          afternoon: this.state.afternoon,
          city: this.state.city,
          current: this.state.current,
          forecast: forecastResult
        })
      });
  }

  filter1Day(dayNumber, forecast){
    return forecast.filter(element => element.FCTTIME.mday === dayNumber);
  }

  daysInside(forecast){
    const notUniques = forecast.map(element => element.FCTTIME.mday);
    let uniques = [];
    for (let i=0; i < notUniques.length; i++){
      if(!uniques.includes(notUniques[i])){
        uniques.push(notUniques[i]);
      }
    }
    return uniques;
  }

  displayRow(weekday_name, morningTemp, morningIcon, afternoonTemp, afternoonIcon, bikeIcon){
    return (
      <tr>
        <td>{weekday_name}</td>
        <td><img class="card-img-top decisionImage" src={bikeIcon} alt="Bike image cap"/></td>
        <td>{morningTemp}°<br/>
            <img class="card-img-top icon" src={morningIcon} alt="Morning Weather icon" />
        </td>
        <td>{afternoonTemp}°<br/>
            <img class="card-img-top icon" src={afternoonIcon} alt="Afternoon Weather icon" />
        </td>
      </tr>
    )
  }

  filterHours(forecast){
    return forecast.filter(hourForecast => hourForecast.FCTTIME.hour === this.state.morning || hourForecast.FCTTIME.hour === this.state.afternoon );
  }

  removeLastDayIfNotFull(forecast){
    if(forecast[forecast.length - 1].FCTTIME.mday !== forecast[forecast.length - 2].FCTTIME.mday){
      forecast.pop();
    }
    return forecast;
  }

  removeFirstDayIfNotFull(forecast){
    if(forecast[0].FCTTIME.mday !== forecast[1].FCTTIME.mday){
      forecast.shift();
    }
    return forecast;
  }

  displayForecast(){
    if(this.state.forecast === "Loading weather forecast"){
      return(
        <tr><td>{this.state.forecast}</td></tr>
      )
    } else {
      let result = this.filterHours(this.state.forecast);
      result = this.removeLastDayIfNotFull(result);
      result = this.removeFirstDayIfNotFull(result);

      console.log("displayForecast");
      console.log(result.map(element => {
        return {
          time: element.FCTTIME.pretty,
          temp: element.temp.metric
        }
      } ));


      return this.daysInside(result).map(dayNumber => {
        const currentDay = this.filter1Day(dayNumber, result);
        return this.displayRow(currentDay[0].FCTTIME.weekday_name, currentDay[0].temp.metric, currentDay[0].icon_url, currentDay[1].temp.metric, currentDay[1].icon_url, this.decideIfBike(currentDay[0], currentDay[1]));
      }
      )
    }
  }

  decideIfBike(dataMorning, dataAfternoon){
    if(dataMorning.fctcode < 8 && dataAfternoon.fctcode < 8 && dataMorning.temp.metric >= 3 && dataAfternoon.temp.metric >=3){ //8 = user limit
      return "http://www.atelierjespers.com/images/pharrell%20williams%20-%20velo%20bleu.jpg";
    }
    return "http://www2.mes-coloriages-preferes.biz/colorino/Images/Large/Vehicules-Voiture-MINI-110563.png";
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to Météo</h1>
        </header>
        <h2>{this.state.city}</h2>
        <h1>{this.state.current}°</h1>
        <table>
          <tr>
            <td></td>
            <td></td>
            <HoursInput morning={this.state.morning} afternoon={this.state.afternoon} handleInputMorning={this.handleInputMorning} handleInputAfternoon={this.handleInputAfternoon}/>
          </tr>
          {this.displayForecast()}
        </table>
      </div>
    );
  }
}

export default App;
