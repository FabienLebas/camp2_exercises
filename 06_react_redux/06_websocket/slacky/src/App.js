import React, { Component } from 'react';
import './App.css';
import Login from './Login.js';
import Chat from './Chat.js';

class App extends Component {
  constructor(props){
    super(props);
    this.state = {
      username:"",
      messages:[]
    };
  }

  componentDidMount(){
    const pop = document.getElementById("pop");
    // We open a connection from the client, related to the server
    const ws = new WebSocket("ws://localhost:4000");
    // What to do when we receive a message?
    ws.onmessage = (event) => {
      console.log("Message: " + event.data);
      const parsedData = JSON.parse(event.data)
      if(parsedData.type === "connection"){
        pop.innerHTML = parsedData.message;
      } else {
        const newMessages = this.state.messages;
        newMessages.push(JSON.parse(event.data));
        this.setState({
          username: this.state.username,
          messages:newMessages
        });
      }
    };
    // Alert the server that the client is gone
    window.addEventListener("beforeunload", () => ws.send("CLOSE"));
  }

  handleLogin = (login) => {
    this.setState({
      username: login,
      messages: this.state.messages
    });
    const ws = new WebSocket("ws://localhost:4000");
    const content = {
      type: "connection",
      message:`${login} is now connected`
    };
    ws.onopen = (event) => {ws.send(JSON.stringify(content));};
  }

  handleLogOut = (login) => {
    this.setState({
      username: "",
      messages:[]
    });
    const ws = new WebSocket("ws://localhost:4000");
    const content = {
      type: "connection",
      message:`${login} disconnected`
    };
    ws.onopen = (event) => {ws.send(JSON.stringify(content));};
  }

  handleMyMessage = (newMessage) => {
    const ws = new WebSocket("ws://localhost:4000");
    ws.onopen = (event) => {ws.send(JSON.stringify(newMessage));};
  }

  render() {
    return (
      <div className="App">
        <div className="jumbotron">
          <Login username={this.state.username} handleLogin={this.handleLogin} handleLogOut={this.handleLogOut}/>
          <p id="pop"></p>
          <Chat username={this.state.username} messages={this.state.messages} handleMyMessage={this.handleMyMessage}/>
        </div>
      </div>
    );
  }
}

export default App;
