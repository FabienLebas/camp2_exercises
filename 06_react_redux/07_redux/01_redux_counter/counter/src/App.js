import React from 'react';
import { connect } from "react-redux";
import './App.css';

const App = (props) => {
  return (
    <div className="App">
      <p className="App-intro">My counter</p>
      <div>{props.counter}</div>
      <button onClick = {props.increment}>+</button>
      <button onClick = {props.decrement}>-</button>
    </div>
  );
}

function mapStateToProps(state) {
  return {
    counter: state
  }
}

function mapDispatchToProps(dispatch) {
  return {
    increment: () => dispatch({type: "INCREMENT"}),
    decrement: () => dispatch({type: "DECREMENT"})
  }
}

const ConnectedApp = connect(mapStateToProps, mapDispatchToProps)(App);

export default ConnectedApp
