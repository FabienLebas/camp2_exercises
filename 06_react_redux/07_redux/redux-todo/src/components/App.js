import React, { Component } from 'react';
import '../App.css';
import Footer from './Footer'
import AddTodo from '../containers/AddTodo'
import VisibleTodoList from '../containers/VisibleTodoList'
import Header from './Header'



class App extends Component {
  render() {
    return (
      <div className="App">
        <Header />
        <AddTodo />
        <VisibleTodoList />
        <Footer />
      </div>
    );
  }
}

export default App;
