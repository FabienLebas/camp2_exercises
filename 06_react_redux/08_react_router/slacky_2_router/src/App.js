import React, { Component } from "react";
import {
  BrowserRouter as Router,
  Route
} from 'react-router-dom';
import "./App.css";
import Login from "./Login";
import Chat from "./Chat";


class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      userName: null,
      messages: [],
    };
    // Attaching the websocket to our App so we can reuse it
    this.websocket = new WebSocket("ws://localhost:8080");
  }
  componentDidMount() {
    // Listen for messages
    this.websocket.addEventListener("message", event => {
      const message = JSON.parse(event.data);
      console.log("Message from server ", message);
      switch (message.type) {
        case "CONNECTION_START":
        default:
          return;
        case "MESSAGES":
          this.setState({ messages: message.data });
          return;
      }
    });
  }

  handleUserName = userName => {
    this.setState({ userName: userName });
    this.websocket.send(
      JSON.stringify({
        type: "LOGIN",
        userName: userName
      })
    );
  };

  sendMessage = message => {
    this.websocket.send(
      JSON.stringify({
        type: "NEW_MESSAGE",
        userName: this.state.userName,
        message: message
      })
    );
  };

  render() {
    return (
      <Router>
        <div className="App">
          <Route exact path="/" render={() => <Login handleUserName={this.handleUserName} userName={this.state.userName}/>}/>
          <Route path="/chat" render={() => <Chat sendMessage={this.sendMessage} messages={this.state.messages} userName={this.state.userName}/>}/>
        </div>
      </Router>
    );
  }
}

export default App;
