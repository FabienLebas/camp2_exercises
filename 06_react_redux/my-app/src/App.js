import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import _ from "underscore";

function displayTableTitle(){
  return (
    <tr>
      <th scope="col" onClick={() => App.sortByModel()}>Modèle</th>
      <th scope="col">Libellé</th>
      <th scope="col">Prix</th>
    </tr>
)  ;
}

class App extends Component {
  constructor(props){
    super(props);
    this.orderNextSort = "asc";
    this.state = {products: props.products};
  }

  sort = (param) => {
    const sortedProducts = _.sortBy(this.state.products, param);
    if(this.orderNextSort === "asc"){
      this.setState({products: sortedProducts});
      this.orderNextSort = "desc";
    } else {
      this.setState({products: sortedProducts.reverse()});
      this.orderNextSort = "asc";
    }
  }

  displayTableRows(myArray){
    return myArray.map(product => (
          <tr>
            <td>{product.decathlon_id}</td>
            <td>{product.title}</td>
            <td>{product.price}</td>
          </tr>
        )
      );
  }

  render() {
    return (
      <div className="App jumbotron">
        <h1 class="text-center">Hello World!</h1>
        <table class="table table-hover">
          <thead>
            <tr>
              <th scope="col" onClick = {() => this.sort("decathlon_id")}>Modèle</th>
              <th scope="col" onClick = {() => this.sort("title")} >Libellé</th>
              <th scope="col" onClick = {() => this.sort("price")}>Prix</th>
            </tr>
          </thead>
          <tbody>
            {this.displayTableRows(this.state.products)}
          </tbody>
        </table>
      </div>
    );
  }
}

export default App;
