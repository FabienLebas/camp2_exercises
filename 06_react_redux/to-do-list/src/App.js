import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import _ from "underscore";

class App extends Component {
  constructor(props){
    super(props);
    this.state = {
      tasks: props.tasks,
      value: ''
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  displayTasks(tasks){
    return tasks.map(task => (
      <tr>
        <td>{task.id}</td>
        <td>{task.title}</td>
        <td><input type="checkbox" onClick = {() => this.remove(task.id)}></input></td>
      </tr>
    ))
  }

  remove(id){
    const removedArray = this.state.tasks.filter(task => task.id !== id);
    this.setState({tasks : removedArray})
  }

  handleChange(event) {
    this.setState({
      tasks: this.state.tasks,
      value: event.target.value});
  }

  handleSubmit(event) {
    event.preventDefault();
    if (!_.isEmpty(this.state.value)){
      const objectToAdd = {
        title: this.state.value,
        id: this.state.tasks.length + 1
      };
      const copy = this.state.tasks;
      copy.push(objectToAdd);
      this.setState({tasks : copy, value: ''});
    }
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to ToDoList</h1>
        </header>
        <div class="jumbotron">
          <form onSubmit={this.handleSubmit}>
            <input type="text" class="form-control" placeholder="Libellé de la tâche" value={this.state.value} onChange={this.handleChange}></input>
            <input type="submit" value="Ajouter une tâche" class="btn btn-success"></input>
          </form>
        </div>
        <table class="table table-hover">
          <thead>
            <tr>
              <td>Numéro</td>
              <td>Titre</td>
              <td>Terminer</td>
            </tr>
          </thead>
          <tbody>
              {this.displayTasks(this.state.tasks)}
          </tbody>
        </table>
      </div>
    );
  }
}

export default App;
