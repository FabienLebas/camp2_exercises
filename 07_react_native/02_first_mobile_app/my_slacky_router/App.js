import React, {Component} from 'react';
import { StyleSheet, Text, View, Button, Image } from 'react-native';
import { DrawerNavigator, NavigationActions } from 'react-navigation';
import Ionicons from 'react-native-vector-icons/Ionicons';
import Login from './src/Login';
import Channels from './src/Channels';
import RootDrawer from './src/RootDrawer';
import App from './src/Home';

class HomeScreen extends Component{
  constructor(props){
    super(props);
    this.state = {
      userName: ""
    }
  }

  authenticate = (userName) => {
    console.log("authenticate");
    this.setState({
      userName: userName
    });
    console.log("authenticate " + this.state.userName);
  }

  render(){
    if(this.state.userName===""){
      return(
          <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
            <Text style={{width:200, fontSize:18, margin:10}}>Welcome to SlackyFab!</Text>
            <Image
              style={{width: 100, height: 100}}
              source={{uri: 'https://seafoammedia.com/wp-content/uploads/2015/01/slack-chat.png'}}
            />
            <Login authenticate={this.authenticate}/>
          </View>
      )
    } else {
      return(
        <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
          <App username={this.state.userName}/>
        </View>
      )
    }
  }
}

export default HomeScreen;
