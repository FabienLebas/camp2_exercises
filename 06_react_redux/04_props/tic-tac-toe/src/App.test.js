import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App key="5"/>, div);
});

it("should look the same", () => {
  const component = renderer.create(
    <App page="http://localhost:3000">Tic Tac Toe</App>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
