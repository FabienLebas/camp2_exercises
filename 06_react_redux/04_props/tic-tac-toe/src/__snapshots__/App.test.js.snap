// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should look the same 1`] = `
<div
  className="App jumbotron"
>
  <nav
    className="navbar fixed-top navbar-dark bg-dark"
  >
    <span
      className="App-title navbar-brand mb-0 h1"
    >
      Tic Tac Toe
    </span>
    <span>
      <button
        className="btn btn-success"
        onClick={[Function]}
      >
        Restart
      </button>
    </span>
  </nav>
  <table>
    <tbody>
      <tr>
        <td
          onClick={[Function]}
        >
          _
        </td>
        <td
          onClick={[Function]}
        >
          _
        </td>
        <td
          onClick={[Function]}
        >
          _
        </td>
      </tr>
      <tr>
        <td
          onClick={[Function]}
        >
          _
        </td>
        <td
          onClick={[Function]}
        >
          _
        </td>
        <td
          onClick={[Function]}
        >
          _
        </td>
      </tr>
      <tr>
        <td
          onClick={[Function]}
        >
          _
        </td>
        <td
          onClick={[Function]}
        >
          _
        </td>
        <td
          onClick={[Function]}
        >
          _
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
